mempool/src/sync/mod.rs

```diff
@@ -0,0 +1,421 @@
+        spawn(fut);
+    }
+
+    /// While there still are unknown transaction hashes which are not part of a request, generate requests and send them to other peers
+    fn send_mempool_transactions_requests(&mut self) {
+        let mut prepared_requests = vec![];
+
+        while self
+            .unknown_hashes
+            .iter()
+            .any(|(_, request_status)| !request_status.is_requested())
+        {
+            for i in 0..self.peers.len() {
+                let peer = self.peers[i];
+                if let Some(request) = self.batch_hashes_by_peer_id(peer) {
+                    prepared_requests.push(request);
+                }
+            }
+        }
+
+        let requests = prepared_requests.into_iter().map(|request| {
+            let peer_id = request.1;
+            let network = Arc::clone(&self.network);
+            async move {
+                (
+                    peer_id,
+                    Self::request_mempool_transactions(
+                        network,
+                        peer_id,
+                        request.0.hashes.to_owned(),
+                    )
+                    .await,
+                )
+            }
+            .boxed()
+        });
+
+        self.transactions_requests.extend(requests);
+    }
+
+    /// Network request for retrieving mempool hashes from other peers
+    async fn request_mempool_hashes(
+        network: Arc<N>,
+        peer_id: N::PeerId,
+        transaction_type: MempoolTransactionType,
+    ) -> Result<ResponseMempoolHashes, RequestError> {
+        network
+            .request::<RequestMempoolHashes>(RequestMempoolHashes { transaction_type }, peer_id)
+            .await
+    }
+
+    /// Network request for retrieving mempool transactions from other peers through a list of provided hashes
+    async fn request_mempool_transactions(
+        network: Arc<N>,
+        peer_id: N::PeerId,
+        hashes: Vec<Blake2bHash>,
+    ) -> Result<ResponseMempoolTransactions, RequestError> {
+        network
+            .request::<RequestMempoolTransactions>(RequestMempoolTransactions { hashes }, peer_id)
+            .await
+    }
+}
+
+impl<N: Network> Stream for MempoolSyncer<N> {
+    type Item = (Transaction, PubsubIdOrPeerId<N>);
+
+    fn poll_next(mut self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Option<Self::Item>> {
+        // First we check if we have a result we can yield
+        if let Some((transaction, peer_id)) = self.transactions.pop_front() {
+            return Poll::Ready(Some((transaction, PubsubIdOrPeerId::PeerId(peer_id))));
+        }
+
+        // Then we check if we should shutdown ourself
+        if self.shutdown_timer.poll_unpin(cx).is_ready() {
+            info!(
+                syncer_type = ?self.mempool_transaction_type,
+                "Shutdown mempool syncer"
+            );
+            return Poll::Ready(None);
+        }
+
+        // Then we check if peers got added to live sync in the mean time
```

Comment:
```suggestion
        // Then we check if peers got added to live sync in the meantime
```
"meantime" has fewer meanings than "mean time".